### ISSUE TITLE
get lua interpreter raise type missmatch in rlm_lua 

### ISSUE THREAD

**Issue by jiangrzh** [FreeRADIUS/freeradius-server#2311]
freeradius-server crash when running with rlm_lua module:
it's throws:
```
Type mismatch: name[char] expected[lua_State *]
````

it's looks like raise on:
```
return *talloc_get_type_abort(marker, lua_State *);
```

**Comment by arr2036** [comment]
Can you give me the actual line number and the steps to reproduce this and I'll take a look.

**Comment by jiangrzh** [comment]
sorry, i was make some change in rlm_lua and migrate it in to 3.0.x
but the code for rlm_lua_get_interp does not change:
```
/** Get a lua interpreter to use
 *
 */
static lua_State *rlm_eslua_get_interp(rlm_eslua_t const *inst) {
    lua_State **marker;

#ifdef HAVE_PTHREAD_H
    lua_State *L;

	/*
	 *	Were running in single interpreter mode, grab the interpreter lock
	 *	and return the instance specific interpreter.
	 */
	if (!inst->threads) {
		if (!inst->interpreter) return NULL;

		pthread_mutex_lock(inst->mutex);
		return inst->interpreter;
	}

	/*
	 *	Were running in multi interpreter mode, retrieve or initialise
	 *	the thread specific, instance specific interpreter.
	 */
	marker = pthread_getspecific(inst->key);
	if (!marker) {
		if (rlm_eslua_init(&L, inst) < 0) {
			return NULL;
		}
		marker = talloc(inst, lua_State *);
		TALLOC_CTX *pool_ctx = talloc_pool(maker, 8*1024*1024);
		talloc_set_destructor(marker, _lua_state_free);
		*marker = L;
		(void) pthread_setspecific(inst->key, marker);

		return L;
	}

	return *talloc_get_type_abort(marker, lua_State *);
#else
    /*
     *	We were build without threads, so inst->threads has no effect.
     *	Just return the instance specific interpreter.
     */
    return inst->interpreter;
#endif
}
```

report is strange:
```
src/modules/rlm_eslua/eslua.c:324: Type mismatch: name[char] expected[lua_State *]
talloc abort: src/modules/rlm_eslua/eslua.c:324: Type mismatch: name[char] expected[lua_State *]
````

**Comment by arr2036** [comment]
I've pushed a fix for v4.0.x.  But the fix was mostly to remove the offending code and use the framework we have in v4.0.x for using thread-specific data.  I'm not really in favour of backporting this code.

I can't stop you, but I like to keep incentives for people to upgrade to newer major versions and help with testing.

If you have other questions about development you're free to ask them on the FreeRADIUS developers list https://freeradius.org/support/.

**Comment by jiangrzh** [comment]
Thank you.
I was write my own lua module in beginning in v3.0.11，but i found some troubles in my code，and i found v4.0.x contains rlm_lua module later . it's fix my problem,so i decide to backporting the rlm_lua module to my verion. 
if v4.0.x released, i will fallow up later.
